### PR TITLE
Bugfix Publication pub type max length

### PR DESCRIPTION
**Pull request by mberacochea**
Truncate the fields in the Publication model if longer than expected.

I've increased the max len of the pub_type in the Publication model.
To prevent the import to fail in the future I've implemented a brute
truncation mechanism when a Publication is saved.

### CODE REVIEW

**Comment by mberacochea** [comment]
Will fix https://www.ebi.ac.uk/panda/jira/browse/EMG-5033

File: emgapi/models.py [hunk 1]
```diff
@@ -502,14 +502,25 @@ class Publication(models.Model):
...
 
     objects = PublicationManager()
 
     class Meta:
         db_table = 'PUBLICATION'
         ordering = ('pubmed_id',)
 
+    def save(self, *args, **kwargs):
+        for field in self._meta.fields:
+            if isinstance(field, models.TextField) or isinstance(field, models.CharField):
```
Comment (mberacochea) [inline emgapi/models.py:515]: We could do something more fancy here, but I would defer that for the time being.

Comment (SandyRogers) [reply emgapi/models.py:515]: Fair. In a future update where we upgrade the db engine, we will probably switch all these artificially max-length'd fields over to `TextField`s anyway.

**Review by SandyRogers — APPROVED** [review]
Thank you!

**Pull request merged**